### PR TITLE
generator: separate option parsing from generator class construction

### PR DESCRIPTION
**Pull request by hilary**
I started work on adding the `--all` option and quickly realized that `command_line.rb` needed some love. So I decided to take a step back and do the refactor in the current version of `command_line.rb` (with the current tests, etc) in a separate PR.

I have not yet made changes to the existing tests. I'd like to get some feedback before proceeding.

### CODE REVIEW

File: lib/generator/command_line.rb [hunk 1]
```diff
@@ -1,26 +1,21 @@
...
       @paths = paths
-      @options = default_options
     end
 
     def parse(args)
-      parse_options(args)
-      generator if valid?
+      parser = GeneratorOptparser.new(args, @paths)
+      @options = parser.parse_options
+      generator if parser.valid?
```
Comment (Insti) [inline lib/generator/command_line.rb]: very minor: We want to know if the options are valid, not if the parser is valid, so this reads oddly to me.

Comment (hilary) [inline lib/generator/command_line.rb]: I'll rename :)

Comment (hilary) [inline lib/generator/command_line.rb]: In fact, the flow needs to be a bit different...

**Comment by kotp** [comment]
Very nice.

**Pull request merged**